### DISCUSSION TITLE
useSimulateContract doesn't return gas estimation

### DISCUSSION THREAD

**Discussion by jd1900** [wevm/wagmi#3596] [Q&A]
In some type of TXs where there is a bit of randomness which makes the gas usage vary, I like to get the gas estimation and increase it by an amount (let's say 1.5 times the simulated value).

On wagmi 1.x I could do it getting the usePrepareContractWrite gas estimation, however, "useSimulateContract" doesn't return the estimated gas. But, I assume that under the hood it should know about it as it is simulation the TX, but it is not being exposed.

Could you expose the gas usage for "useSimulateContract"?

**Comment by jd1900** [comment]
Nothing?

**Comment by generatorpoint** [comment]
There is the `useEstimateGas` hook for this, but I'm not sure why it was removed from the prepare/simulate hooks

**Reply by agsola** [reply]
Using two RPC calls, which doesn't make sense.

**Reply by generatorpoint** [reply]
It appears it was removed to reduce compute unit cost of `useSimulateContract`. 

https://github.com/wevm/wagmi/issues/2367#issuecomment-1552108983

**Reply by agsola** [reply]
Thank you @generatorpoint , I hadn't seen that conversation.
I still think it may be interesting to include it. 
We like to use it to to give the option to increase the gas limit, let's say from estimation to estimation * 1.2.
